webapp/channels/src/components/team_sidebar/components/team_button.tsx

```diff
@@ -36,186 +30,215 @@ interface Props {
     placement?: 'left' | 'right' | 'top' | 'bottom';
     teamIconUrl?: string | null;
     switchTeam: (url: string) => void;
-    intl: IntlShape;
     isDraggable?: boolean;
     teamIndex?: number;
     teamId?: string;
     isInProduct?: boolean;
     hasUrgent?: boolean;
 }
 
-class TeamButton extends React.PureComponent<Props> {
-    handleSwitch = (e: React.MouseEvent) => {
+export default function TeamButton({
+    btnClass,
+    url,
+    displayName,
+    order,
+    unread,
+    mentions,
+    teamIconUrl,
+    isDraggable = false,
+    switchTeam,
+    teamIndex,
+    teamId,
+    tip,
+    ...props
+}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleSwitch = useCallback((e: React.MouseEvent) => {
         mark('TeamLink#click');
         e.preventDefault();
-        this.props.switchTeam(this.props.url);
+        switchTeam(url);
 
         setTimeout(() => {
             trackEvent('ui', 'ui_team_sidebar_switch_team');
         }, 0);
-    };
+    }, [switchTeam, url]);
 
-    render() {
-        const {teamIconUrl, displayName, btnClass, mentions, unread, isDraggable = false, teamIndex, teamId, order} = this.props;
-        const {formatMessage} = this.props.intl;
+    let teamClass: string = props.active ? 'active' : '';
```

Comment:
There is a lot of logic around teamClass, and I wonder if it would be simpler to use `classnames`. But 0/5.

For sure, this would look way better next to the rest of the `teamClass` logic :P

Reply:
Yeah, it's be a bit nicer, but I'm not going to bother with it now since it's just a minor style thing